### ISSUE TITLE
bug(input): not enabled

### ISSUE THREAD

**Issue by OneOfOne** [folke/snacks.nvim#239]
### Did you check docs and existing issues?

- [X] I have read all the snacks.nvim docs
- [X] I have updated the plugin to the latest version before submitting this issue
- [X] I have searched the existing issues of snacks.nvim
- [X] I have searched the existing issues of plugins related to this issue

### Neovim version (nvim -v)

NVIM v0.11.0-dev-1326+g3bb2d02759

### Operating system/version

Arch / 6.12.3-2-cachyos

### Describe the bug

Enabling the input snack doesn't actually make it work, checkhealth shows:

```
Snacks.input ~
- OK setup {enabled}
- ERROR `vim.ui.input` is not set to `Snacks.input`
```

### Steps To Reproduce

enable the input snack

### Expected Behavior

Use it instead of the default vim.ui.input

### Repro

```lua
vim.env.LAZY_STDPATH = ".repro"
load(vim.fn.system("curl -s https://raw.githubusercontent.com/folke/lazy.nvim/main/bootstrap.lua"))()

require("lazy.minit").repro({
  spec = {
    { "folke/snacks.nvim", opts = {} },
    -- add any other plugins here
  },
})
```

**Comment by daphen** [comment]
I'm still having the same problem, I'm not sure how to enable this.

**Comment by folke** [comment]
If you still get that health check error, then iot means you or another plugin is overriding `vim.ui.input`.
Nothing I can do about it.

**Comment by RileyMathews** [comment]
I'm also seeing this error and the checkhealth is reporting that input is set to `Snacks.input`. On the most recent version as of this comment.
![image](https://github.com/user-attachments/assets/ea366ca8-1032-4337-9fd3-8d67384545c3)

**Comment by daphen** [comment]
> I'm also seeing this error and the checkhealth is reporting that input is set to `Snacks.input`. On the most recent version as of this comment. ![image](https://private-user-images.githubusercontent.com/37124104/397804240-ea366ca8-1032-4337-9fd3-8d67384545c3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ3MTMyNTYsIm5iZiI6MTczNDcxMjk1NiwicGF0aCI6Ii8zNzEyNDEwNC8zOTc4MDQyNDAtZWEzNjZjYTgtMTAzMi00MzM3LTlmZDMtOGQ2NzM4NDU0NWMzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjIwVDE2NDIzNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIwY2M1YTMzY2E2ZDkwOWUwOTBlNWQwMmQ1MjNkZTQyMGU0NTVhOWVhYWEzN2QwYzQ5OTc3OTZmYjliYmE1OTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.kUYV23cyzXZy2Rpa5bRqoEF4r1Eix6gMKaWmsriwTA8)

Yeah that's my current status as well. I can't get it to work so I'm using Noice for the time being, the input from nui that's set by noice is working perfectly.

**Comment by daphen** [comment]
> And why do you think input is not working? It looks pretty much the same as the one in noice. Do `esc` once then you should go in `normal` mode. With the noice one, that's not possible.

Well, because when I remove Noice and enable input in Snacks the ui is at the bottom and nothing happens even though checkhealth says it's loaded ok and that UI is set to snacks.input.

**Comment by folke** [comment]
How do you test it? You need to test it with `vim.ui.input`, not `vim.fn.input`.
Noice does `vim.fn.input` which is also the default impl for `vim.ui.input`.

**Comment by daphen** [comment]
> How do you test it? You need to test it with `vim.ui.input`, not `vim.fn.input`. Noice does `vim.fn.input` which is also the default impl for `vim.ui.input`.

I am not experienced enough in Neovim to fully be able to debug things that aren't working unfortunately. I just expect my command input to be a floating input box instead of the default one in the bottom left after enabling snacks.input. Do you have something quick that I can do to test it? ELI5 unfortunately.

**Comment by RileyMathews** [comment]
FWIW I also tried disabling all plugins except for snacks and lazy and still had this issue. I am in the same boat where I am still a bit of a novice when it comes to debugging neovim internals and customization. Aside from plugins and other keymappings which I also tried temporarily commenting out my neovim config is basically just a handful of plugins and keymappings.

**Comment by daphen** [comment]
Same boat here. Everything disabled including keymaps and custom options, only Snacks enabled.

**Comment by folke** [comment]
This is not an **issue**.
In Neovim, there has always been `vim.fn.input`, to get input from the user on the cmdline.
Later they added `vim.ui.input`, which allows plugins to get input asynchronously.
The default implementation of that in Neovim is just `vim.fn.input`.

Plugins can override `vim.ui.input` (but not `vim.fn.input`) to provide a better user experience.

That's what `snacks.input` does.

Noice is a different beast. Since noice implements `ui_attach`, it can override the whole ui, including the cmdline and as such also `vim.fn.input`, but still with the same limitations as before, namely that it has to be synchronous.

So again, this is **NOT** a bug. You just don't understand what `vim.ui.input` is for.

Please just read the Neovim help to learn more.

**Comment by daphen** [comment]
> This is **not a bug**.
> 
> In Neovim, there has always been `vim.fn.input`, to get input from the user on the cmdline (synchronously). Later they added `vim.ui.input`, which allows plugins to get input asynchronously. The default implementation of that in Neovim is just `vim.fn.input`.
> 
> Plugins can override `vim.ui.input` (but not `vim.fn.input`) to provide a better user experience.
> 
> That's what `snacks.input` does.
> 
> Noice is a different beast. Since noice implements `ui_attach`, it can override the whole ui, including the cmdline and as such also `vim.fn.input`, but still with the same limitations as before, namely that it has to be synchronous.
> 
> So again, this is **NOT** a bug. You just don't understand what `vim.ui.input` is for.
> 
> Please read the Neovim help to learn more.

Alright, thanks for the clarification!

**Comment by folke** [comment]
to clarify, modern plugins all use `vim.ui.input` to get input from the user.
older plugins, or vim plugins won't be using this.

**Comment by daphen** [comment]
And vim.ui.input is not what is triggered when a user presses `:` ?

**Comment by folke** [comment]
No, not at all, that's the cmdline.
Has nothing to do with `vim.ui.input` or `vim.fn.input`.

You can test with for example: `:lua Snacks.rename.rename_file()`

**Comment by daphen** [comment]
> No, not at all, that's the cmdline. Has nothing to do with `vim.ui.input` or `vim.fn.input`.
> 
> You can test with for example: `:lua Snacks.rename.rename_file()`

Haha, that explains why it's "not working". 😅
Sorry for wasting your time.
